src/main/java/dev/jbang/JdkManager.java

```diff
@@ -0,0 +1,75 @@
+package dev.jbang;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class JdkManager {
+	private static final String JDK_DOWNLOAD_URL = "https://api.adoptopenjdk.net/v3/binary/latest/%d/ga/%s/%s/jdk/hotspot/normal/adoptopenjdk";
+
+	public static Path getCurrentJdk(String requestedVersion) throws IOException {
+		int currentVersion = JavaUtil.determineJavaVersion();
+		int actualVersion = JavaUtil.javaVersion(requestedVersion);
+		if (currentVersion == actualVersion) {
+			return JavaUtil.getJdkHome();
+		} else {
+			return getInstalledJdk(actualVersion);
+		}
+	}
+
+	public static Path getInstalledJdk(int version) throws IOException {
+		Path jdkDir = getJdkPath(version);
+		if (!Files.isDirectory(jdkDir)) {
+			jdkDir = downloadAndInstallJdk(version, false);
+		}
+		return jdkDir;
+	}
+
+	public static Path downloadAndInstallJdk(int version, boolean updateCache) throws IOException {
+		System.err.println("Downloading JDK " + version + "...");
+		String url = String.format(JDK_DOWNLOAD_URL, version, Util.getOS().name(), Util.getArch().name());
+		Path jdkPkg = Util.downloadAndCacheFile(url, updateCache);
+		Path jdkDir = getJdkPath(version);
+		try {
+			System.err.println("Installing JDK " + version + "...");
+			UnpackUtil.unpack(jdkPkg, jdkDir);
+		} catch (Throwable th) {
+			// Make sure we don't leave halfway unpacked files around
+			Util.deleteFolder(jdkDir, true);
+			throw th;
+		}
+		return jdkDir;
+	}
+
+	public static Set<Integer> listInstalledJdks() throws IOException {
+		if (Files.isDirectory(getJdksPath())) {
+			Supplier<TreeSet<Integer>> sset = () -> new TreeSet<>();
+			return Files.list(getJdksPath()).map(p -> {
+				try {
+					return Integer.parseInt(p.getFileName().toString());
+				} catch (NumberFormatException ex) {
+					return -1;
+				}
+			}).filter(v -> v > 0).collect(Collectors.toCollection(sset));
+		} else {
+			return Collections.emptySet();
+		}
+	}
+
+	public static boolean isInstalledJdk(int version) {
+		return Files.isDirectory(getJdkPath(version));
+	}
+
+	private static Path getJdkPath(int version) {
+		return getJdksPath().resolve(Integer.toString(version));
```

Comment:
i'm wondering if we should make it `<cache>/jdk/<version>/default/` or similar for now? in case we want to support `//JAVA 11.0.4` notion and then don't have a folder to store that in ?

Reply:
Not sure, with what intention exactly? In case we want multiple v14 JDKs installed? Couldn't we than just switch to naming the directories with their full versions? Choosing the correct folder could easily be done in code.

Reply:
you would scan the folder for "latest" 14 version for the default version?

Reply:
Well, what is "the default version"? Either you let people decide (which might mean creating a `current` link, but at what level? Just the `current` global Java version or the current 14 version and have possibly multiple "current"s?) or you indeed scan for the latest,

Personally, I wouldn't mind taking on SDKMAN a bit here and let people define a single "current" version, eg: `jbang sdk default 14.0.2`, we add a `jbang jdk home` that returns the path to the current version (which uses a link so will always be `~/.jbang/cache/jdk/current`). People can use that to add `export PATH=$(jbang jdk home):$PATH` to their `.bashrc` files.